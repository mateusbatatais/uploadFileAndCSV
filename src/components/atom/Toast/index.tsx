interface Props {
  msg: string;
  state: boolean;
  type: string;
}

const Toast = ({ msg, state, type }: Props) => {
  return (
    <div
      className={`${state ? "" : "d-none "}alert mt-3 alert-${type}`}
      role="alert"
    >
      {msg}
    </div>
  );
};

export default Toast;
