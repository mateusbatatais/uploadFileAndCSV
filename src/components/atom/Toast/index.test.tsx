import { render, screen } from "@testing-library/react";
import Toast from "./index";

describe("toast component", () => {
  test("correct render text", () => {
    render(<Toast msg="Text dasd message" type="success" state={true} />);
    const linkElement = screen.getByText(/Text message/i);
    expect(linkElement).toBeInTheDocument();
  });

  test("snapshot", () => {
    const { container } = render(
      <Toast msg="Text message" type="success" state={true} />
    );
    expect(container).toMatchSnapshot();
  });
});
