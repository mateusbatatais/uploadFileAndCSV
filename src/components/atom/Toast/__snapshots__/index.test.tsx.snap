// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toast component snapshot 1`] = `
<div>
  <div
    class="alert mt-3 alert-success"
    role="alert"
  >
    Text message
  </div>
</div>
`;
